### ISSUE TITLE
Connection problem to kernel

### ISSUE THREAD

**Issue by Ashirov** [microsoft/vscode-jupyter#9020]
## Environment data

-   VS Code version: 1.64.2
-   Jupyter Extension version: v2022.1.1201831736
-   Python Extension version: v2022.0.1814523869
-   OS (Windows) version: 21H1
-   Python version: 3.8.10
-   Jupyter server running: Local

## Actual behaviour

Connecting to kernel: Python 3.8.10 64-bit: Waiting for Jupyter Session to be idle

## Logs

Visual Studio Code (1.64.2, undefined, desktop)
Jupyter Extension Version: 2022.1.1201831736.
Python Extension Version: 2022.0.1814523869.
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'jupyterEnhancedDataViewer'
Info 13:25:26: ZMQ install verified.
Info 13:25:27: Preferred kernel connection found in cache .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 13:25:27: PreferredConnection: .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher found for NotebookDocument: c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:27: Early registration of controller for Kernel connection .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 13:25:27: TargetController found ID: .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher for document c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:27: Setting controller affinity for c:\Python\Exercises\Untitled-1.ipynb .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 13:25:27: Interrupt requested & no kernel.
Info 13:25:27: Setting setActiveController for c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:27: Getting activation commands for C:\Python\Python38\python.exe are not cached. May take a while.
Info 13:25:27: KernelProvider switched kernel to id = .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 13:25:27: Starting Notebook in kernel.ts id = .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 13:25:27: Creating raw notebook for c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:27: Getting preferred kernel for c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:27: Computing working directory c:\Python\Exercises\Untitled-1.ipynb
Info 13:25:28: Experiment status for python is {"enabled":true,"optInto":[],"optOutFrom":[]}
Info 13:25:28: Starting raw kernel Python 3.8.10 64-bit
Info 13:25:28: installMissingDependencies C:\Python\Python38\python.exe
Info 13:25:28: IPykernel found previously in this environment C:\Python\Python38\python.exe
Info 13:25:28: Kernel launching with ports 9000,9001,9002,9003,9004. Start port is 9000
Info 13:25:28: No custom variables for Kernel as interpreter is not conda, but is Global
Info 13:25:30: Process Execution: > C:\Python\Python38\python.exe -m pip list
> C:\Python\Python38\python.exe -m pip list
Info 13:25:30: Process Execution: > C:\Python\Python38\python.exe -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
> C:\Python\Python38\python.exe -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
Info 13:25:30: Creating controller for jupyter-notebook with interpreter C:\Python\Python38\python.exe
Info 13:25:30: Creating daemon process for C:\Python\Python38\python.exe with env variables count 0
Info 13:25:30: Process Execution: > C:\Python\Python38\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_interrupt_daemon -v --ppid 5644
> C:\Python\Python38\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_interrupt_daemon -v --ppid 5644
Info 13:25:31: get interrupthandle daemon
Info 13:25:31: Process Execution: > C:\Python\Python38\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"eb59f091-2896-4d2c-bf66-6318e45175e3" --shell=9002 --transport="tcp" --iopub=9004 --f=C:\Users\AsusPro\AppData\Local\Temp\tmp-56446hFbCe6qn3Re.json
> C:\Python\Python38\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"eb59f091-2896-4d2c-bf66-6318e45175e3" --shell=9002 --transport="tcp" --iopub=9004 --f=C:\Users\AsusPro\AppData\Local\Temp\tmp-56446hFbCe6qn3Re.json
Info 13:25:31: Process Execution: cwd: c:\Python\Exercises
cwd: c:\Python\Exercises
Info 13:25:31: Registering dummy command feature
Info 13:25:34: ipykernel version 5.4.2 for C:\Python\Python38\python.exe
Info 13:25:34: ipykernel location ~\AppData\Roaming\Python\Python38\site-packages\ipykernel\__init__.py for C:\Python\Python38\python.exe
Warn 13:25:34: StdErr from Kernel Process C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\traitlets\traitlets.py:2196: FutureWarning: Supporting extra quotes around Unicode is deprecated in traitlets 5.0. Use 'hmac-sha256' instead of '"hmac-sha256"' – or use CUnicode.
  warn(

Warn 13:25:34: StdErr from Kernel Process C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\traitlets\traitlets.py:2151: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use 'eb59f091-2896-4d2c-bf66-6318e45175e3' instead of 'b"eb59f091-2896-4d2c-bf66-6318e45175e3"'.
  warn(

Info 13:25:34: Kernel Output: NOTE: When using the `ipython kernel` entry point, Ctrl-C will not work.

To exit, you will have to explicitly quit this process, by either sending
"quit" from a client, or using Ctrl-\ in UNIX-like environments.

To read more about this, see https://github.com/ipython/ipython/issues/2049

To connect another client to this kernel, use:
    --existing C:\Users\AsusPro\AppData\Local\Temp\tmp-56446hFbCe6qn3Re.json

Info 13:25:35: Raw session started and connected
Started kernel Python 3.8.10 64-bit
Info 13:25:35: Finished connecting 03a373af-aac6-49e4-a175-09d5228afe57
Warn 13:25:36: StdErr from Kernel Process [IPKernelApp] ERROR | Exception in message handler:
Traceback (most recent call last):
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\kernelbase.py", line 265, in dispatch_shell
    yield gen.maybe_future(handler(stream, idents, msg))
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\tornado\gen.py", line 762, in run
    value = future.result()
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\tornado\gen.py", line 234, in wrapper
    yielded = ctx_run(next, result)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\kernelbase.py", line 580, in complete_request
    matches = yield gen.maybe_future(self.do_complete(code, cursor_pos))
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\ipkernel.py", line 356, in do_complete
    return self._experimental_do_complete(code, cursor_pos)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\ipkernel.py", line 381, in _experimental_do_complete
    completions = list(_rectify_completions(code, raw_completions))
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\IPython\core\completer.py", line 484, in rectify_completions
    completions = list(completions)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\IPython\core\completer.py", line 1818, in completions
    for c in self._completions(text, offset, _timeout=self.jedi_compute_type_timeout/1000):
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\IPython\core\completer.py", line 1861, in _completions
    matched_text, matches, matches_origin, jedi_matches = self._complete(
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\IPython\core\completer.py", line 2029, in _complete
    completions = self._jedi_matches(
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\IPython\core\completer.py", line 1373, in _jedi_matches
    interpreter = jedi.Interpreter(
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\jedi\api\__init__.py", line 725, in __init__
    super().__init__(code, environment=environment,
TypeError: __init__() got an unexpected keyword argument 'column'

**Comment by DonJayamanne** [comment]
Thanks for filing this issue, please could you provide the following information:
* output from the command `python -m pip list` (where `python` points to your Python38 environment)
* Please could you run the command `python -m pip install ipykernel ipython jedi parso pyzmq -U --force-reinstall`
* Reload VS Code and let me know how that goes
* If it still doesn't work, please provide the output from `python -m pip list` again & the output from the `Jupyter` output panel

**Comment by Ashirov** [comment]
- **output from the command `python -m pip list` (where `python `points to your Python38 environment):**

PS C:\Python\Python38> .\python.exe -m pip list
Package           Version
----------------- -------
astroid           2.4.2
autopep8          1.6.0
backcall          0.2.0
colorama          0.4.4
cycler            0.10.0
decorator         4.4.2
et-xmlfile        1.0.1
ipykernel         5.4.2
ipython           7.19.0
ipython-genutils  0.2.0
isort             5.6.4
jdcal             1.4.1
jedi              0.18.0
joblib            1.0.0
jupyter-client    6.1.7
jupyter-core      4.7.0
kiwisolver        1.3.1
lazy-object-proxy 1.4.3
matplotlib        3.3.3
mccabe            0.6.1
mlxtend           0.18.0
numpy             1.19.4
openpyxl          3.0.5
pandas            1.1.5
parso             0.8.1
pickleshare       0.7.5
Pillow            8.0.1
pip               21.1.1
prompt-toolkit    3.0.8
pycodestyle       2.8.0
Pygments          2.7.3
pylint            2.6.0
pyparsing         2.4.7
python-dateutil   2.8.1
pytz              2020.4
pywin32           300
pyzmq             20.0.0
scikit-learn      0.23.2
scipy             1.5.4
setuptools        56.0.0
six               1.15.0
sklearn           0.0
threadpoolctl     2.1.0
toml              0.10.2
tornado           6.1
traitlets         5.0.5
wcwidth           0.2.5
wrapt             1.12.1
xlrd              1.2.0
WARNING: You are using pip version 21.1.1; however, version 22.0.3 is available.
You should consider upgrading via the 'C:\Python\Python38\python.exe -m pip install --upgrade pip' command.



**- Please could you run the command `python -m pip install ipykernel ipython jedi parso pyzmq -U --force-reinstall`:**

PS C:\Python\Python38> python -m pip install ipykernel ipython jedi parso pyzmq -U --force-reinstall
Collecting ipykernel
  Downloading ipykernel-6.9.0-py3-none-any.whl (128 kB)
     |████████████████████████████████| 128 kB 544 kB/s
Collecting ipython
  Downloading ipython-8.0.1-py3-none-any.whl (747 kB)
     |████████████████████████████████| 747 kB 656 kB/s
Collecting jedi
  Downloading jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)
     |████████████████████████████████| 1.6 MB 1.6 MB/s
Collecting parso
  Downloading parso-0.8.3-py2.py3-none-any.whl (100 kB)
     |████████████████████████████████| 100 kB 1.3 MB/s
Collecting pyzmq
  Downloading pyzmq-22.3.0-cp38-cp38-win_amd64.whl (1.0 MB)
     |████████████████████████████████| 1.0 MB 1.1 MB/s
Collecting jupyter-client<8.0
  Downloading jupyter_client-7.1.2-py3-none-any.whl (130 kB)
     |████████████████████████████████| 130 kB 1.6 MB/s
Collecting tornado<7.0,>=4.2
  Using cached tornado-6.1-cp38-cp38-win_amd64.whl (422 kB)
Collecting debugpy<2.0,>=1.0.0
  Downloading debugpy-1.5.1-cp38-cp38-win_amd64.whl (4.4 MB)
     |████████████████████████████████| 4.4 MB 2.2 MB/s
Collecting traitlets<6.0,>=5.1.0
  Downloading traitlets-5.1.1-py3-none-any.whl (102 kB)
     |████████████████████████████████| 102 kB 2.2 MB/s
Collecting matplotlib-inline<0.2.0,>=0.1.0
  Downloading matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)
Collecting nest-asyncio
  Downloading nest_asyncio-1.5.4-py3-none-any.whl (5.1 kB)
Collecting decorator
  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting colorama
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting black
  Downloading black-22.1.0-cp38-cp38-win_amd64.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB 2.2 MB/s
Collecting setuptools>=18.5
  Downloading setuptools-60.9.1-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 3.3 MB/s
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting pygments
  Downloading Pygments-2.11.2-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 3.3 MB/s
Collecting stack-data
  Downloading stack_data-0.2.0-py3-none-any.whl (21 kB)
Collecting prompt-toolkit!=3.0.0,!=3.0.1,<3.1.0,>=2.0.0
  Downloading prompt_toolkit-3.0.28-py3-none-any.whl (380 kB)
     |████████████████████████████████| 380 kB 3.2 MB/s
Collecting entrypoints
  Downloading entrypoints-0.4-py3-none-any.whl (5.3 kB)
Collecting jupyter-core>=4.6.0
  Downloading jupyter_core-4.9.1-py3-none-any.whl (86 kB)
     |████████████████████████████████| 86 kB 1.9 MB/s
Collecting python-dateutil>=2.1
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     |████████████████████████████████| 247 kB 3.3 MB/s
Collecting pywin32>=1.0
  Downloading pywin32-303-cp38-cp38-win_amd64.whl (9.3 MB)
     |████████████████████████████████| 9.3 MB 3.3 MB/s
Collecting wcwidth
  Using cached wcwidth-0.2.5-py2.py3-none-any.whl (30 kB)
Collecting six>=1.5
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting tomli>=1.1.0
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting platformdirs>=2
  Downloading platformdirs-2.5.0-py3-none-any.whl (14 kB)
Collecting click>=8.0.0
  Downloading click-8.0.3-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB 2.3 MB/s
Collecting mypy-extensions>=0.4.3
  Downloading mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting pathspec>=0.9.0
  Downloading pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Collecting typing-extensions>=3.10.0.0
  Downloading typing_extensions-4.1.1-py3-none-any.whl (26 kB)
Collecting asttokens
  Downloading asttokens-2.0.5-py2.py3-none-any.whl (20 kB)
Collecting pure-eval
  Downloading pure_eval-0.2.2-py3-none-any.whl (11 kB)
Collecting executing
  Downloading executing-0.8.2-py2.py3-none-any.whl (16 kB)
Installing collected packages: six, colorama, wcwidth, typing-extensions, traitlets, tomli, pywin32, pure-eval, platformdirs, pathspec, parso, mypy-extensions, executing, click, asttokens, tornado, stack-data, setuptools, pyzmq, python-dateutil, pygments, prompt-toolkit, pickleshare, nest-asyncio, matplotlib-inline, jupyter-core, jedi, entrypoints, decorator, black, backcall, jupyter-client, ipython, debugpy, ipykernel
  Attempting uninstall: six
    Found existing installation: six 1.15.0
    Uninstalling six-1.15.0:
      Successfully uninstalled six-1.15.0
  Attempting uninstall: colorama
    Found existing installation: colorama 0.4.4
    Uninstalling colorama-0.4.4:
      Successfully uninstalled colorama-0.4.4
  Attempting uninstall: wcwidth
    Found existing installation: wcwidth 0.2.5
    Uninstalling wcwidth-0.2.5:
      Successfully uninstalled wcwidth-0.2.5
  Attempting uninstall: traitlets
    Found existing installation: traitlets 5.0.5
    Uninstalling traitlets-5.0.5:
      Successfully uninstalled traitlets-5.0.5
  Attempting uninstall: pywin32
    Found existing installation: pywin32 300
    Uninstalling pywin32-300:
ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'c:\\users\\asuspro\\appdata\\roaming\\python\\python38\\site-packages\\win32\\win32api.pyd'
Consider using the `--user` option or check the permissions.


**- Reload VS Code and let me know how that goes:**
_The same.. still can not connect to the kernel. I don't know if the previous command did its job properly. It gave me an error saying "**Could not install packages due to an OSError**" as I have pasted the output above._

**- If it still doesn't work, please provide the output from python -m pip list again:**

PS C:\Python\Python38> .\python.exe -m pip list
WARNING: Ignoring invalid distribution -ywin32 (c:\users\asuspro\appdata\roaming\python\python38\site-packages)
Package           Version
----------------- -------
astroid           2.4.2  
autopep8          1.6.0  
backcall          0.2.0  
colorama          0.4.4  
cycler            0.10.0 
decorator         4.4.2  
et-xmlfile        1.0.1  
ipykernel         5.4.2  
ipython           7.19.0 
ipython-genutils  0.2.0  
isort             5.6.4  
jdcal             1.4.1  
jedi              0.18.0 
joblib            1.0.0  
jupyter-client    6.1.7  
jupyter-core      4.7.0  
kiwisolver        1.3.1  
lazy-object-proxy 1.4.3  
matplotlib        3.3.3  
mccabe            0.6.1  
mlxtend           0.18.0 
numpy             1.19.4 
openpyxl          3.0.5  
pandas            1.1.5
parso             0.8.1
pickleshare       0.7.5
Pillow            8.0.1
pip               21.1.1
prompt-toolkit    3.0.8
pycodestyle       2.8.0
Pygments          2.7.3
pylint            2.6.0
pyparsing         2.4.7
python-dateutil   2.8.1
pytz              2020.4
pyzmq             20.0.0
scikit-learn      0.23.2
scipy             1.5.4
setuptools        56.0.0
six               1.16.0
sklearn           0.0
threadpoolctl     2.1.0
toml              0.10.2
tomli             2.0.1
tornado           6.1
traitlets         5.1.1
typing-extensions 4.1.1
wcwidth           0.2.5
wrapt             1.12.1
xlrd              1.2.0
WARNING: Ignoring invalid distribution -ywin32 (c:\users\asuspro\appdata\roaming\python\python38\site-packages)
WARNING: Ignoring invalid distribution -ywin32 (c:\users\asuspro\appdata\roaming\python\python38\site-packages)
WARNING: Ignoring invalid distribution -ywin32 (c:\users\asuspro\appdata\roaming\python\python38\site-packages)
WARNING: You are using pip version 21.1.1; however, version 22.0.3 is available.
You should consider upgrading via the 'C:\Python\Python38\python.exe -m pip install --upgrade pip' command.

**- the output from the Jupyter output panel:**

Visual Studio Code (1.64.2, undefined, desktop)
Jupyter Extension Version: 2022.1.1201831736.
Python Extension Version: 2022.0.1814523869.
User belongs to experiment group 'jupyterTest'
User belongs to experiment group 'jupyterEnhancedDataViewer'
Info 15:56:37: ZMQ install verified.
Info 15:56:38: Preferred kernel connection found in cache .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 15:56:38: PreferredConnection: .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher found for NotebookDocument: c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Early registration of controller for Kernel connection .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 15:56:38: TargetController found ID: .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher for document c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Setting controller affinity for c:\Python\Exercises\Untitled-1.ipynb .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 15:56:38: Interrupt requested & no kernel.
Info 15:56:38: Setting setActiveController for c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Getting activation commands for C:\Python\Python38\python.exe are not cached. May take a while.
Info 15:56:38: KernelProvider switched kernel to id = .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 15:56:38: Starting Notebook in kernel.ts id = .jvsc74a57bd0e404b59586357c814bc0d3940e75d6763c00a48753b225b81f7716971b8e1741.C:\Python\Python38\python.exe.C:\Python\Python38\python.exe.-m#ipykernel_launcher
Info 15:56:38: Creating raw notebook for c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Getting preferred kernel for c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Computing working directory c:\Python\Exercises\Untitled-1.ipynb
Info 15:56:38: Starting raw kernel Python 3.8.10 64-bit
Info 15:56:38: installMissingDependencies C:\Python\Python38\python.exe
Info 15:56:38: IPykernel found previously in this environment C:\Python\Python38\python.exe
Info 15:56:38: Kernel launching with ports 9000,9001,9002,9003,9004. Start port is 9000
Info 15:56:38: No custom variables for Kernel as interpreter is not conda, but is Global
Info 15:56:38: Experiment status for python is {"enabled":true,"optInto":[],"optOutFrom":[]}
Info 15:56:40: Process Execution: > C:\Python\Python38\python.exe -m pip list
> C:\Python\Python38\python.exe -m pip list
Info 15:56:40: Process Execution: > C:\Python\Python38\python.exe -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
> C:\Python\Python38\python.exe -c "import ipykernel; print(ipykernel.__version__); print("5dc3a68c-e34e-4080-9c3e-2a532b2ccb4d"); print(ipykernel.__file__)"
Info 15:56:40: Creating controller for jupyter-notebook with interpreter C:\Python\Python38\python.exe
Info 15:56:41: Creating daemon process for C:\Python\Python38\python.exe with env variables count 0
Info 15:56:41: Process Execution: > C:\Python\Python38\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_interrupt_daemon -v --ppid 2092
> C:\Python\Python38\python.exe -m vscode_datascience_helpers.daemon --daemon-module=vscode_datascience_helpers.kernel_interrupt_daemon -v --ppid 2092
Info 15:56:42: get interrupthandle daemon
Info 15:56:42: Process Execution: > C:\Python\Python38\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"99f7b585-3016-4443-a8bd-ae14ef661cba" --shell=9002 --transport="tcp" --iopub=9004 --f=C:\Users\AsusPro\AppData\Local\Temp\tmp-20926WdrpqvoE770.json
> C:\Python\Python38\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"99f7b585-3016-4443-a8bd-ae14ef661cba" --shell=9002 --transport="tcp" --iopub=9004 --f=C:\Users\AsusPro\AppData\Local\Temp\tmp-20926WdrpqvoE770.json
Info 15:56:42: Process Execution: cwd: c:\Python\Exercises
cwd: c:\Python\Exercises
Info 15:56:42: Registering dummy command feature
Info 15:56:44: ipykernel version 5.4.2 for C:\Python\Python38\python.exe
Info 15:56:44: ipykernel location ~\AppData\Roaming\Python\Python38\site-packages\ipykernel\__init__.py for C:\Python\Python38\python.exe
Warn 15:56:45: StdErr from Kernel Process C:\Python\Python38\lib\site-packages\traitlets\traitlets.py:2202: FutureWarning: Supporting extra quotes around strings is deprecated in traitlets 5.0. You can use 'hmac-sha256' instead of '"hmac-sha256"' if you require traitlets >=5.
  warn(

Warn 15:56:45: StdErr from Kernel Process C:\Python\Python38\lib\site-packages\traitlets\traitlets.py:2157: FutureWarning: Supporting extra quotes around Bytes is deprecated in traitlets 5.0. Use '99f7b585-3016-4443-a8bd-ae14ef661cba' instead of 'b"99f7b585-3016-4443-a8bd-ae14ef661cba"'.
  warn(

Warn 15:56:45: StdErr from Kernel Process Traceback (most recent call last):

Warn 15:56:45: StdErr from Kernel Process   File "C:\Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel_launcher.py", line 16, in <module>
    app.launch_new_instance()
  File "C:\Python\Python38\lib\site-packages\traitlets\config\application.py", line 845, in launch_instance
    app.initialize(argv)
  File "C:\Python\Python38\lib\site-packages\traitlets\config\application.py", line 88, in inner
    return method(app, *args, **kwargs)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\kernelapp.py", line 570, in initialize
    self.write_connection_file()
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\ipykernel\kernelapp.py", line 230, in write_connection_file
    write_connection_file(cf, ip=self.ip, key=self.session.key, transport=self.transport,
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\jupyter_client\connect.py", line 138, in write_connection_file
    with secure_write(fname) as f:
  File "C:\Python\Python38\lib\contextlib.py", line 113, in __enter__
    return next(self.gen)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\jupyter_core\paths.py", line 461, in secure_write
    win32_restrict_file_to_user(fname)
  File "C:\Users\AsusPro\AppData\Roaming\Python\Python38\site-packages\jupyter_core\paths.py", line 387, in win32_restrict_file_to_user
    import win32api
ModuleNotFoundError: No module named 'win32api'

Error 15:56:45: Disposing session as kernel process died ExitCode: 1, Reason: ModuleNotFoundError
Info 15:56:45: Dispose Kernel process
Info 15:56:45: kill daemon
Warn 15:57:05: Didn't get response for requestKernelInfo after 20021ms.
Info 15:57:05: Raw session started and connected
Started kernel Python 3.8.10 64-bit
Info 15:57:05: Finished connecting 7f1d5e6f-1fe1-4459-b43a-ddffab678b58

**Comment by Ashirov** [comment]
Now it give this error
![image](https://user-images.githubusercontent.com/46714728/154213699-32fa6d6b-72a4-4cb4-8f1b-c58d95e98f49.png)

**Comment by DonJayamanne** [comment]
@Ashirov 
Please could you install the latest pre-release version of the jupyter extesnsion.
& please install the python package `win32api`  (click on `learn more` for details)
If it still doesn't work please create a new issue (as you are now getting different errors).
Please note, I would like to get you unblocked but request you create a new issue as the errors are now unrelated, and helps when we deal with them in a separate issue.

**Comment by Ashirov** [comment]
I installed the pre-release version (v2022.2.1010480428) and it works now.

**Comment by DonJayamanne** [comment]
Duplicate of #9014
